src/main/java/com/google/devtools/build/lib/buildtool/AnalysisPhaseRunner.java

```diff
@@ -214,6 +214,7 @@ private static AnalysisResult runAnalysisPhase(
             loadingResult,
             targetOptions,
             multiCpu,
+            request.getTargets(),
```

Comment:
Re: your comment about matching `:foo` with `//:foo`, `//mypkg:foo`, etc:

I agree that `AnalysisPhaseRunner.evaluateTargetPatterns` has the most relevant logic. How about writing your own helper method here that rewrites `request.getTargets()` with the following logic extracted from `evaluateTargetPatterns`:

```
TargetPattern.Parser parser =
    new TargetPattern.Parser(env.getRelativeWorkingDirectory());
if (parser.parse(rawPattern).getType() == TargetPattern.SINGLE_TARGET) {
  /* This is an explicit request */
String fullLabel = parsedPattern.getSingleTargetPath();
}
```

When I try this this returns "@//mypkg:foo`. I'm not sure if you'll want to remove the `@` part? And with this approach I guess `requestedTargetPatterns` just becomes `explicitTargetPatterns`?

Reply:
That worked beautifully, thank you!
`TargetPattern` was the key. Really makes it simple :)

Reply:
I also added a test case to make sure this works with external repositories.